### PR TITLE
Add reserve increasing

### PR DESCRIPTION
**Pull request by dylandesrosier**

### CODE REVIEW

File: src/PrizePool.sol [hunk 1]
```diff
@@ -55,6 +55,12 @@ error SmoothingGTEOne(int64 smoothing);
 /// @param available The available un-accounted balance that can be claimed as a contribution
 error ContributionGTDeltaBalance(uint256 amount, uint256 available);
 
+/// @notice Emitted when a user attempts to increase the reserve amount by more than their balance.
+/// @param user The user attempting to increase the reserve
+/// @param balance The users current balance
```
Comment (trmid) [inline src/PrizePool.sol:60]:
```suggestion
/// @param balance The user's current balance
```

**Review by trmid — APPROVED** [review]
small grammar change

File: src/PrizePool.sol [hunk 2]
```diff
@@ -55,6 +55,12 @@ error SmoothingGTEOne(int64 smoothing);
 /// @param available The available un-accounted balance that can be claimed as a contribution
 error ContributionGTDeltaBalance(uint256 amount, uint256 available);
 
+/// @notice Emitted when a user attempts to increase the reserve amount by more than their balance.
+/// @param user The user attempting to increase the reserve
+/// @param balance The user's current balance
+/// @param amount The amount that the reserve is being increased by
+error InsufficientPrizeTokenBalance(address user, uint256 balance, uint256 amount);
```
Comment (asselstine) [inline src/PrizePool.sol:62]: Where did this come from?

File: src/PrizePool.sol [hunk 3]
```diff
@@ -167,6 +173,10 @@ contract PrizePool is TieredLiquidityDistributor {
   /// @param amount The amount of assets transferred
   event WithdrawReserve(address indexed to, uint256 amount);
 
+  /// @notice Emitted when the reserve is manually increased.
+  /// @param amount The amount of assets transferred
+  event IncreaseReserve(uint256 amount);
```
Comment (asselstine) [inline src/PrizePool.sol:178]: Should prob include the user that increased?

**Review by asselstine — CHANGES_REQUESTED** [review]

**Review by asselstine — APPROVED** [review]

**Pull request merged**